src/ai/assistant.ts

```diff
@@ -0,0 +1,440 @@
+        {
+          id: 'asst_CswXI1p9YGKr32Ks2ENPb5VL',
+          model: 'gpt-4',
+          name: 'AINA-EBAJYJF111',
+          instructions: '',
+          description: 'OpenAI의 가장 강력한 모델입니다.',
+          metadata: {
+            image: 'https://picsum.photos/id/2/200/200',
+          },
+          created_at: 1711969423,
+        },
+        {
+          id: 'asst_IfWuJqqO5PdCF9DbgZRcFClG',
+          model: 'gpt-3.5-turbo',
+          name: 'AINA-TKAJYJF1C5',
+          instructions: '',
+          description: '일상적인 작업에 적합합니다. GPT-3.5-turbo에 의해 구동됩니다.',
+          metadata: {
+            image: 'https://picsum.photos/id/1/200/200',
+          },
+          created_at: 1704034800,
+        },
+      ],
+    };
+    */
+  }
+
+  async mintAndCreate(
+    objectId: string,
+    to: string,
+    { model, name, instructions, description, metadata }: AssistantCreateParams
+  ): Promise<AssistantMinted> {
+    const checksum = getChecksumAddress(to);
+
+    await validateObject(this.ain, objectId);
+
+    const serverName = this.ainize.getServerName();
+    await validateServerConfigurationForObject(this.ain, objectId, serverName);
+
+    const opType = OperationType.MINT_CREATE_ASSISTANT;
+    const body = {
+      objectId,
+      to: checksum,
+      model,
+      name,
+      instructions,
+      ...(description && { description }),
+      ...(metadata && !_.isEmpty(metadata) && { metadata }),
+    };
+
+    const { data } = await this.ainize.requestWithAuth<AssistantMinted>(this.ain, {
+      serverName,
+      opType,
+      data: body,
+    });
+
+    return data;
+    // NOTE(jiyoung): example data
+    /*
+    return {
+      tokenId: '1',
+      objectId: '0xCE3c4D8dA38c77dEC4ca99cD26B1C4BF116FC401',
+      owner: '0x7ed9c30C9F3A31Daa9614b90B4a710f61Bd585c0',
+      assistant: {
+        id: 'asst_IfWuJqqO5PdCF9DbgZRcFClG',
+        model: 'gpt-3.5-turbo',
+        name: 'AINA-TKAJYJF1C5',
+        instructions: '',
+        description: '일상적인 작업에 적합합니다. GPT-3.5-turbo에 의해 구동됩니다.',
+        metadata: {
+          image: 'https://picsum.photos/id/1/200/200',
+        },
+        created_at: 1704034800,
+      },
+    };
+    */
+  }
+
+  private buildTxBodyForCreateAssistant(
+    { id, model, name, instructions, description, metadata, created_at }: Assistant,
+    objectId: string,
+    tokenId: string,
+    address: string
+  ) {
+    const appId = AinftObject.getAppId(objectId);
+    const assistantPath = Path.app(appId).token(tokenId).ai().value();
+    const historyPath = `/apps/${appId}/tokens/${tokenId}/ai/history/$user_addr`;
+    const threadPath = `${historyPath}/threads/$thread_id`;
+    const messagePath = `${threadPath}/messages/$message_id`;
+
+    const config = {
+      model,
+      name,
+      instructions,
+      ...(description && { description }),
+      ...(metadata && !_.isEmpty(metadata) && { metadata }),
+    };
+    const info = {
+      id,
+      type: 'chat',
+      config,
+      createdAt: created_at,
```

Comment:
@platfowner I added two fields to assistant schema in blockchain. (type, createdAt)
- type: 
   - 'chat', (in future, 'image', 'multi-modal', 'voice'..)
   - type of assistant is determined by the specialized functions of the model.
   - ref) https://huggingface.co/models
- createdAt:
   -  need for retrieving and sorting assistant list information from blockchain data.